src/storage/txn/scheduler.rs

```diff
@@ -1492,13 +1494,30 @@ impl Scheduler {
             group.push(cid);
         }
     }
+}
+
+/// A Msg Receiver wrapper to impl Display so that Scheduler Runnable can be impl-ed
+pub struct MsgReceiver {
+    pub inner: Receiver<Msg>,
+}
+
+impl Display for MsgReceiver {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "")
+    }
+}
 
-    pub fn run(&mut self, receiver: Receiver<Msg>) -> Result<()> {
+impl Runnable<MsgReceiver> for Scheduler {
+    fn run(&mut self, receiver: MsgReceiver) {
```

Comment:
I am confused that why not using run_batch directly, but still using a channel to receive messages?

Reply:
Currently `scheduler.run()` this function is a event loop that runs in a different thread, when we want to execute some command, we can't directly call `scheduler.run` or `scheduler.run_batch`, we need the channel to talk to the scheduler.
I am trying to think of a way to remove this event loop, so that maybe we can remove channel.

Reply:
of course, we should remove the scheduler origin channel and use worker directly.

Reply:
Move this channel mechanism into `worker` instead `scheduler` itself, Am I right?

Reply:
no. Worker already has Channel, so remove the channel from scheduler. 

You can ask @hicqu for help when using Worker.

Reply:
yeah, that's what I mean, the channel in `Scheduler` becomes channel in `Worker`.

Reply:
+1